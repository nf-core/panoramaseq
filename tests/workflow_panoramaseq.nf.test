nextflow_workflow {

    name "Test PANORAMA_SEQ workflow"
    script "workflows/panorama_seq/main.nf"
    workflow "panorama_seq"

    tag "workflow"
    tag "workflow_panoramaseq"

    test("panoramaseq workflow - test profile") {

        when {
            params {
                outdir = "$outputDir"
                
                // Test data
                star_genome_dir = params.pipelines_testdata_base_path + 'reference/star_index'
                star_gtf = params.pipelines_testdata_base_path + 'reference/genes.gtf'
                mergecounts = false
                validate_h5ad = true
            }
            
            workflow {
                """
                input[0] = Channel.fromList([
                    [
                        [ 
                            id: 'sample1',
                            N_barcodes: 1000,
                            barcode_file: params.pipelines_testdata_base_path + 'reference/barcodes_coords_test.csv',
                            sample_size: 100,
                            len_barcode: 25,
                            Nthresh: 5,
                            Ntriage: 50,
                            single_end: false
                        ],
                        [
                            params.pipelines_testdata_base_path + 'fastq/sample1_R1.fastq.gz',
                            params.pipelines_testdata_base_path + 'fastq/sample1_R2.fastq.gz'
                        ]
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.multiqc_report).match("multiqc") },
                { assert snapshot(workflow.out.versions).match("versions") }
            )
        }
    }
}
