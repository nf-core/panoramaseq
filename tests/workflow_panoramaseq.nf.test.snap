{
  "panoramaseq workflow - test profile": {
    "content": [
      {
        "0": "multiqc",
        "1": "versions"
      }
    ],
    "meta": {
      "nextflow": "24.04.2",
      "nf-test": "0.9.2"
    },
    "timestamp": "2025-06-20T10:00:00.000000000"
  }
}
